tutorial/admin/test_prereqs.sh

```diff
@@ -0,0 +1,213 @@
+    echo "       Please read the \"USER ENVIRONMENT\" section of the main README."
+    RC=1
+elif ! 2>/dev/null ${PYTHON_CMD} -c "import geopmpy.launcher"; then
+    echo "ERROR: The following Python dependencies are missing:"
+    2>&1 ${PYTHON_CMD} -c "import geopmpy.launcher" | grep Module
+    echo "       Please read the \"PYTHON INSTALL\" section of the main README."
+    RC=1
+elif ! 2>/dev/null ${PYTHON_CMD} -c "import geopmpy.analysis"; then
+    echo "ERROR: The following Python dependencies are missing:"
+    2>&1 ${PYTHON_CMD} -c "import geopmpy.analysis" | grep Module
+    echo "       Please read the \"PYTHON INSTALL\" section of the main README."
+    RC=1
+fi
+
+# SYSTEMD Config check
+if [ "$(basename $(realpath /sbin/init))" = "systemd" ]; then
+    # Using SYSTEMD
+    CONF=/etc/systemd/logind.conf
+    if [ "$(grep --quiet ^RemoveIPC ${CONF} | cut -d= -f2)" = "yes" ]; then
+        echo "ERROR: systemd is set improperly and will cause shared memory"
+        echo "       issues with GEOPM.  Set RemoveIPC=no in ${CONF}"
+        echo "       Please read the \"SYSTEMD CONFIGURATION\" section of the main README."
+        RC=1
+    elif [ "$(grep --quiet ^#RemoveIPC ${CONF} | cut -d= -f2)" = "yes" ]; then
+        echo "ERROR: The compile time defaults for systemd will cause issues"
+        echo "       with GEOPM.  Set RemoveIPC=no in ${CONF}"
+        echo "       Please read the \"SYSTEMD CONFIGURATION\" section of the main README."
+        RC=1
+    elif ! grep --quiet RemoveIPC ${CONF}; then
+        echo "WARNING: Unable to determine RemoveIPC value from ${CONF}"
+        echo "         Run the GEOPM unit tests to ensure there are no issues"
+        echo "         with shared memory.  SharedMemoryTest.fd_check"
+        echo "         Please read the \"SYSTEMD CONFIGURATION\" section of the main README."
+    fi
+else
+    echo "WARNING: Your system is not using systemd.  Please run the GEOPM
+                   unit tests to ensure there are no issues with shared memory.
+                   Specifically: SharedMemoryTest.fd_check"
+fi
+
+# msr-safe configuration
+MSRSAVE=/usr/sbin/msrsave
+if command_available_and_executable ${MSRSAVE}; then
+    REQUIRED_VERSION=1.4.0
+    INSTALLED_VERSION=$(${MSRSAVE} --version | head -n1 | cut -d- -f1)
+    if [ "${REQUIRED_VERSION}" != "$(echo -e "${REQUIRED_VERSION}\n${INSTALLED_VERSION}" | sort -V | head -n1)" ]; then
+        echo "ERROR: Installed version of msr-safe (${INSTALLED_VERSION}) is too old."
+        echo "       Please update msr-safe to at least version ${REQUIRED_VERSION}."
+        RC=1
+    fi
+else
+    echo "WARNING: Unable to find/execute ${MSRSAVE}.  Unable to determine msr-safe version."
+fi
+
+FILE=/dev/cpu/msr_batch
+if [ ! -c ${FILE} ]; then
+    echo "ERROR: ${FILE} is not available.  Please install msr-safe."
+    RC=1
+fi
+
+if ! ${MSRSAVE} /tmp/msrsave_test; then
+    echo "ERROR: ${MSRSAVE} had a non-zero return code."
```

Comment:
msrsave unit test failed.  What are they expected to do?   probably should contact GEOPM or msr-safe maintainers

Reply:
This should also check for existence of msrsave executable

Reply:
The existence check is on line 91.